File: frontend/src/api/types/enums.ts
/**
 * Backend enum types.
 * Values mirror the Python Enum members in app/schemas/enums.py exactly.
 * DO NOT add values that do not exist in the backend.
 */

/** app/schemas/enums.py – Department */
export type Department = 'Engineering' | 'Traction' | 'S&T';

/** app/schemas/enums.py – DefectSeverity */
export type DefectSeverity = 'A' | 'B' | 'C';

/** app/schemas/enums.py – Weekday */
export type Weekday = 'Mon' | 'Tue' | 'Wed' | 'Thu' | 'Fri' | 'Sat' | 'Sun';

/** app/schemas/enums.py – PlanningHorizonType */
export type PlanningHorizonType = 'daily' | 'weekly' | 'monthly';

/** app/schemas/enums.py – TaskStatus */
export type TaskStatus = 'open' | 'scheduled' | 'unscheduled' | 'completed' | 'cancelled';

/** app/schemas/enums.py – PlanStatus */
export type PlanStatus = 'pending' | 'generating' | 'ready' | 'failed' | 'superseded';

/** app/schemas/enums.py – AssignmentStatus */
export type AssignmentStatus = 'scheduled' | 'unscheduled' | 'overridden';

/** app/schemas/enums.py – OverrideAction */
export type OverrideAction = 'reassign' | 'force_schedule' | 'unschedule';

/** app/schemas/enums.py – ErrorCode */
export type ErrorCode =
  | 'validation_error'
  | 'not_found'
  | 'conflict'
  | 'dependency_not_ready'
  | 'plan_generation_failed'
  | 'override_rejected';

File: backend/app/schemas/enums.py
"""Explicit domain enumerations for the SIH26027 API contract."""

from enum import Enum


class Department(str, Enum):
    """Maintenance departments in the current prototype datasets."""

    ENGINEERING = "Engineering"
    TRACTION = "Traction"
    S_AND_T = "S&T"


class DefectSeverity(str, Enum):
    """Ordinal defect classes used by scoring (A is most severe)."""

    A = "A"
    B = "B"
    C = "C"


class Weekday(str, Enum):
    """Weekday labels used by the prototype window and schedule CSVs."""

    MON = "Mon"
    TUE = "Tue"
    WED = "Wed"
    THU = "Thu"
    FRI = "Fri"
    SAT = "Sat"
    SUN = "Sun"


class PlanningHorizonType(str, Enum):
    DAILY = "daily"
    WEEKLY = "weekly"
    MONTHLY = "monthly"


class TaskStatus(str, Enum):
    """Lifecycle of a maintenance request in the planning API."""

    OPEN = "open"
    SCHEDULED = "scheduled"
    UNSCHEDULED = "unscheduled"
    COMPLETED = "completed"
    CANCELLED = "cancelled"


class PlanStatus(str, Enum):
    PENDING = "pending"
    GENERATING = "generating"
    READY = "ready"
    FAILED = "failed"
    SUPERSEDED = "superseded"


class AssignmentStatus(str, Enum):
    SCHEDULED = "scheduled"
    UNSCHEDULED = "unscheduled"
    OVERRIDDEN = "overridden"


class OverrideAction(str, Enum):
    REASSIGN = "reassign"
    FORCE_SCHEDULE = "force_schedule"
    UNSCHEDULE = "unschedule"


class ErrorCode(str, Enum):
    VALIDATION_ERROR = "validation_error"
    NOT_FOUND = "not_found"
    CONFLICT = "conflict"
    DEPENDENCY_NOT_READY = "dependency_not_ready"
    PLAN_GENERATION_FAILED = "plan_generation_failed"
    OVERRIDE_REJECTED = "override_rejected"
